scenario_player/utils/token.py

```diff
@@ -15,456 +17,249 @@
-        return self.config.spaas.rpc.client_id
-
-    @property
-    def address(self) -> HexAddress:
-        return self._address
-
-    @property
-    def balance(self):
-        return self._local_rpc_client.balance(self.address)
-
-    @property
-    def checksum_address(self) -> ChecksumAddress:
-        """Checksum'd address of the deployed contract."""
-        return to_checksum_address(self.address)
-
-    def transact(self, action: str, parameters: dict) -> TransactionHash:
-        """Send a transact request to `/rpc/contract/<action>` and return the resulting tx hash."""
-        payload = {
-            "client_id": self.client_id,
-            "gas_limit": self.config.gas_limit,
-            "contract_address": self.checksum_address,
-        }
-        payload.update(parameters)
-
-        log.info(f"Requesting '{action}' call", **payload)
-        resp = self.interface.post(f"spaas://rpc/contract/{action}", json=payload)
-        resp.raise_for_status()
-        resp_data = resp.json()
-        tx_hash = resp_data["tx_hash"]
-        log.info(f"'{action}' call succeeded", tx_hash=tx_hash)
-        return tx_hash
-
-    def mint(
-        self, target_address, required_balance=None, max_fund_amount=None, **kwargs
-    ) -> Union[TransactionHash, None]:
-        """Mint new tokens for the given `target_address`.
-
-        The amount of tokens depends on the scenario definition's settings, and defaults to
-        :attr:`.DEFAULT_TOKEN_BALANCE_MIN` and :attr:`.DEFAULT_TOKEN_BALANCE_FUND`
-        if those settings are absent.
-        """
-        local_log = log.bind(contract=self.name)
-        balance = self.balance
-        if required_balance is None:
-            required_balance = self.config.token.min_balance
-        local_log.debug(
-            "Checking necessity of mint request",
-            required_balance=required_balance,
-            actual_balance=balance,
-        )
-        if not balance < required_balance:
-            local_log.debug("Mint call not required - sufficient funds")
-            return
+TokenDetails = TypedDict("TokenDetails", {"name": str, "address": ChecksumAddress, "block": int})
 
-        if max_fund_amount is None:
-            max_fund_amount = self.config.token.max_funding
 
-        mint_amount = max_fund_amount - balance
-        local_log.debug("Minting required - insufficient funds.", mint_amount=mint_amount)
-        params = {"amount": mint_amount, "target_address": target_address}
-        params.update(kwargs)
-        return self.transact("mint", params)
+def token_maybe_mint(
+    token_proxy: Token, target_address, minimum_balance: int, maximum_balance: int
+) -> Optional[TransactionHash]:
+    current_balance = token_proxy.balance(target_address)
 
+    if minimum_balance > current_balance:
+        mint_amount = maximum_balance - current_balance
+        return token_proxy.mint(amount=mint_amount, target_address=target_address)
 
-class Token(Contract):
```

Comment:
All of this code was duplicated and removed in favor of `raiden.network.proxies.token.Token`